### PR TITLE
Bump Routing to 5.6, include Middleware

### PR DESCRIPTION
**Pull request by AkenRoberts**
_Note: this branch is rebased on PR #66 so there's some duplication. Can adjust if needed._

* Routing component bumped to Laravel 5.6. No additional changes were required.
* Middleware examples were merged into Routing component, and are no longer separated. All code is up-to-date for 5.6.

### CODE REVIEW

**Comment by Rubemlrm** [comment]
@AkenRoberts hi there, in the normal cases we only update this project components to LTS releases , like 5.1 and 5.5. Since 5.6 is not a LTS release maybe its better to "wait". But @mattstauffer can provide a more correct information relative to this. Besides that thanks a lot for the PR.

**Comment by AkenRoberts** [comment]
Thanks for the note! Following the LTS release makes sense - I could see arguments for both.

Routing and middleware didn't really change from 5.5 to 5.6. I'd be happy to make the minor adjustment to follow the LTS version if desired.

**Comment by mattstauffer** [comment]
So what I've traditionally done is, for every LTS, done a big update where the new LTS is the "current" version. Obviously it's past time for 5.5

So let's make this one 5.5. Meanwhile I'll split off a 5.1 branch like I did with the 4.2 branch so we can have the "new" content all be 5.5.

**Comment by mattstauffer** [comment]
So what I've traditionally done is, for every LTS, done a big update where the new LTS is the "current" version. Obviously it's past time for 5.5

So let's make this one 5.5. Meanwhile I'll split off a 5.1 branch like I did with the 4.2 branch so we can have the "new" content all be 5.5.

**Comment by mattstauffer** [comment]
Whoops. Wait. We already did that. LOL. Then yes, please just update this to 5.5 and we can merge. lol. My bad.

**Comment by Rubemlrm** [comment]
@mattstauffer we made that a few months ago eheh. Btw if you need help to maintain this project just say , i have some free time and can help you maintain this.

**Comment by AkenRoberts** [comment]
Rebased changes now in master and reverted back to 5.5 LTS.

File: components/routing/readme.md [hunk 1]
```diff
@@ -4,10 +4,11 @@
 
 # Routing
 
-This component shows how to use Laravel's [Routing](https://laravel.com/docs/5.5/routing) features in non-Laravel applications.
```
Comment (m1guelpf) [inline components/routing/readme.md]: Update doc version here

**Comment by mattstauffer** [comment]
@Rubemlrm lol yep i caught myself after like a minute; just left the old comment around so you could all see how dumb i am.

File: components/middleware/composer.json [hunk 2]
```diff
@@ -1,11 +0,0 @@
-{
```
Comment (mattstauffer) [inline components/middleware/composer.json]: What's the benefit of merging this with routing?

File: components/routing/composer.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
     "require": {
-        "illuminate/routing": "^5.5",
-        "illuminate/events": "^5.5"
+        "illuminate/routing": "5.5.*",
+        "illuminate/events": "5.5.*"
     },
     "autoload": {
         "psr-4": {
-            "App\\": "src/"
+            "Torch\\Routing\\": "src/"
```
Comment (mattstauffer) [inline components/routing/composer.json]: I prefer `App`

File: components/routing/src/Controllers/DashboardController.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Torch\Routing\Controllers;
+
+class DashboardController
+{
+    public function login(): string
```
Comment (mattstauffer) [inline components/routing/src/Controllers/DashboardController.php]: No need for return type hints

File: components/routing/src/Middleware/Authenticate.php [hunk 5]
```diff
@@ -16,8 +16,8 @@ class Authenticate
      */
     public function handle($request, Closure $next, $guard = null)
     {
-        if (! $_SESSION['user']) {
-            return 'Error Authenticate. Please <a href="/login">login</a>';
+        if (! @$_SESSION['logged_in']) {
```
Comment (mattstauffer) [inline components/routing/src/Middleware/Authenticate.php]: I don't like `@`; what are you covering for here? Can you replace it with `isset`?

File: components/routing/src/Middleware/RedirectIfAuthenticated.php [hunk 6]
```diff
@@ -15,10 +12,10 @@ class RedirectIfAuthenticated
      * @param  string|null  $guard
      * @return mixed
      */
-    public function handle($request, Closure $next, $guard = null)
+    public function handle($request, \Closure $next, $guard = null)
```
Comment (mattstauffer) [inline components/routing/src/Middleware/RedirectIfAuthenticated.php]: I prefer not using `\` inline, but instead moving it to the use block above

File: components/routing/src/Middleware/StartSession.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  string|null  $guard
+     * @return mixed
+     */
+    public function handle($request, \Closure $next, $guard = null)
```
Comment (mattstauffer) [inline components/routing/src/Middleware/StartSession.php]: Please `use` the Closure; and I don't believe `session_start()` needs the preceding slash

File: components/middleware/composer.json (hunk 2 above)
Comment (AkenRoberts) [reply components/middleware/composer.json]: Is middleware not a feature of routing? I don't see when or how someone would use Laravel middleware and not routing.

Also, the code examples for both were almost identical. Middleware just happened to include middleware. You could break out the middleware part into a separate PHP file within routing, but I would still keep it all contained in the Routing parent.

File: components/routing/composer.json (hunk 3 above)
Comment (AkenRoberts) [reply components/routing/composer.json]: I made a note in the commit description about why I did this. PhpStorm warns about duplicate class definitions if more than one component has the same FQCN. It doesn't functionally impact the code, but that was why.

File: components/routing/src/Middleware/Authenticate.php (hunk 5 above)
Comment (AkenRoberts) [reply components/routing/src/Middleware/Authenticate.php]: `isset` would fix this, yes. In the event you don't access the protected routes in the correct order, the `logged_in` index won't exist and you'd get a notice.

File: components/routing/src/Middleware/StartSession.php (hunk 7 above)
Comment (AkenRoberts) [reply components/routing/src/Middleware/StartSession.php]: `session_start()` doesn't _need_ the preceding class, it's just a micro-optimization that I think is a good standard.

File: components/middleware/composer.json (hunk 2 above)
Comment (mattstauffer) [reply components/middleware/composer.json]: I'm fine if it's separated within routing, but it was separated for a reason: the vast majority of people who use Laravel and want to use its routing elsewhere don't fully understand middleware. The goal of this project is to make people's learning curve and ease-of-porting-elsewhere as low as possible; that's why they were originally separate, if i remember correctly. So, separate file within routing is cool! But let's at least make it separate.

File: components/routing/composer.json (hunk 3 above)
Comment (mattstauffer) [reply components/routing/composer.json]: Thanks--I'd still prefer `App`, and storm can just deal with it. Thanks!

File: components/routing/src/Middleware/Authenticate.php [hunk 8]
```diff
@@ -16,8 +16,8 @@ class Authenticate
      */
     public function handle($request, Closure $next, $guard = null)
     {
-        if (! $_SESSION['user']) {
-            return 'Error Authenticate. Please <a href="/login">login</a>';
+        if (isset($_SESSION['logged_in']) && ! $_SESSION['logged_in']) {
```
Comment (mattstauffer) [inline components/routing/src/Middleware/Authenticate.php]: Could you explain this change (logged_in with the isset and the check vs just user)? Thanks!

File: readme.md [hunk 9]
```diff
@@ -20,7 +20,7 @@ Now you can visit [http://localhost:8000/](http://localhost:8000/) in your brows
  * [Config](https://github.com/mattstauffer/Torch/tree/master/components/config)
  * [Encryption](https://github.com/mattstauffer/Torch/tree/master/components/encryption)
  * [Log](https://github.com/mattstauffer/Torch/tree/master/components/log)
- * [Routing](https://github.com/mattstauffer/Torch/tree/master/components/routing)
+ * [Routing + Middleware](https://github.com/mattstauffer/Torch/tree/master/components/routing)
```
Comment (mattstauffer) [inline readme.md]: Let's do `&` instead of `+`

**Pull request closed without merging**